pxtlib/tilemap.ts

```diff
@@ -677,6 +677,98 @@ namespace pxt {
+            if (asset.meta?.blockIDs?.length > 0) return true;
+
+            if (asset.type == pxt.AssetType.Tile) {
+                for (const tm of this.getAssets(AssetType.Tilemap)) {
+                    if (tm.data.tileset.tiles.some(t => t.internalID === asset.internalID)) {
+                        return true;
+                    }
+                }
+            }
+
+            if (files) {
+                const shortId = getShortIDForAsset(asset);
+                const displayName = asset.meta?.displayName || "";
+
+                let assetTsRefs: string;
+                switch (asset.type) {
+                    case pxt.AssetType.Tile:
+                        assetTsRefs = `myTiles.${shortId}|assets.tile\`${shortId}\``;
+                        if (displayName) assetTsRefs += `|assets.tile\`${displayName}\``;
+                        break;
+                    case pxt.AssetType.Tilemap:
+                        assetTsRefs = `tilemap\`${shortId}\``;
+                        break;
+                    case pxt.AssetType.Animation:
+                        assetTsRefs = `assets.animation\`${shortId}\``;
+                        if (displayName) assetTsRefs += `|assets.animation\`${displayName}\``;
+                        break;
+                    default:
+                        assetTsRefs = `assets.image\`${shortId}\``;
+                        if (displayName) assetTsRefs += `|assets.image\`${displayName}\``;
+                        break;
+                }
+                const assetTsRegex = new RegExp(assetTsRefs, "gm");
+
+                let assetPyRefs: string;
+                switch (asset.type) {
+                    case pxt.AssetType.Tile:
+                        assetPyRefs = `myTiles.${shortId}|assets.tile\("""${shortId}"""\)`;
+                        if (displayName) assetPyRefs += `|assets.tile\("""${displayName}"""\)`;
+                        break;
+                    case pxt.AssetType.Tilemap:
+                        assetPyRefs = `assets.tilemap\("""${shortId}"""\)`;
+                        break;
+                    case pxt.AssetType.Animation:
+                        assetPyRefs = `assets.animation\("""${shortId}"""\)`;
+                        if (displayName) assetPyRefs += `|assets.animation\("""${displayName}"""\)`;
+                        break;
+                    default:
+                        assetPyRefs = `assets.image\("""${shortId}"""\)`;
+                        if (displayName) assetPyRefs += `|assets.image\("""${displayName}"""\)`;
+                        break;
+                }
+                const assetPyRegex = new RegExp(assetPyRefs, "gm");
+
+                for (let filename of Object.keys(files)) {
+                    const f = files[filename];
+                    // Match .ts files that are not generated (.g.ts)
+                    if (filename.match(/(?<!\.g)\.ts$/i)) {
```

Comment:
Ahh wait a second, forgot about this; safari doesn't support lookbehind :( https://caniuse.com/mdn-javascript_builtins_regexp_lookbehind_assertion